numerics/frequency_analysis_body.hpp

```diff
@@ -46,7 +149,100 @@ AngularFrequency PreciseMode(
   return GoldenSectionSearch(amplitude,
                              fft_mode.min,
                              fft_mode.max,
-                             std::greater<Square<DotResult>>());
+                             std::greater<Square<Value>>());
+}
+
+template<typename Function,
+         int degree_, int wdegree_,
+         template<typename, typename, int> class Evaluator>
+PoissonSeries<std::invoke_result_t<Function, Instant>, degree_, Evaluator>
+Projection(
+    AngularFrequency const& ω,
+    Function const& function,
+    PoissonSeries<double, wdegree_, Evaluator> const& weight,
+    DotProduct<Function, std::invoke_result_t<Function, Instant>,
+               degree_, wdegree_, Evaluator> const& dot) {
+  using Value = std::invoke_result_t<Function, Instant>;
+  using Series = PoissonSeries<Value, degree_, Evaluator>;
+
+  Instant const& t0 = weight.origin();
+  auto const basis = BasisGenerator<Series>::Basis(ω, t0);
+  constexpr int basis_size = std::tuple_size_v<decltype(basis)>;
+
+  // This code follows [Kud07], section 2.  Our indices start at 0, unlike those
+  // of Кудрявцев which start at 1.
+  FixedLowerTriangularMatrix<Inverse<Value>, basis_size> α;
+  std::vector<Function> f;
+
+  // Only indices 0 to m - 1 are used in this array.  At the beginning of
+  // iteration m it contains Aⱼ⁽ᵐ⁻¹⁾.
+  std::array<double, basis_size> A;
+
+  auto const F₀ = dot(function, basis[0], weight);
+  // TODO(phl): This does not work if basis does not have the same type as
+  // Function, i.e., if the degrees don't match.
+  auto const Q₀₀ = dot(basis[0], basis[0], weight);
+  α[0][0] = 1 / Sqrt(Q₀₀);
+  A[0] = F₀ / Q₀₀;
+  f.emplace_back(function - A[0] * basis[0]);
+  for (int m = 1; m < basis_size; ++m) {
+    // Contains Fₘ.
+    auto const F = dot(f[m - 1], basis[m], weight);
+
+    // Only indices 0 to m are used in this array.  It contains Qₘⱼ.
+    std::array<Square<Value>, basis_size> Q;
+    for (int j = 0; j <= m; ++j) {
+      Q[j] = dot(basis[m], basis[j], weight);
+    }
+
+    // Only indices 0 to m - 1 are used in this array.  It contains Bⱼ⁽ᵐ⁾.
+    std::array<Value, basis_size> B;
+    for (int j = 0; j < m; ++j) {
+      Value accumulator;
```

Comment:
Could we call this `Σ_αⱼₛ_Qₘₛ` (with `+=` and a `-` below)?
We do that in many places (`Σ_Bₗ_m_yˡ` in the elliptic integrals, `Σj_aʹ_ij_g_jk` in an integrator (we should switch to subscripts there someday), etc.), and `accumulator` is not very informative.

Same for the other accumulators.

This would make it much easier to check the code against the paper.

Reply:
Will do.  (Tofu, tofu, tofu, tofus tofui tofu, tofua tofua tofua, tofuum tofibus tofibus.)